libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -155,6 +210,19 @@ bool StaticAnalyzer::visit(VariableDeclaration const& _variable)
 			// This is not a no-op, the entry might pre-exist.
 			m_localVarUseCount[make_pair(_variable.id(), &_variable)] += 0;
 	}
+
+	// This excludes struct members on purpose.
+	if (_variable.isStateVariable() || _variable.referenceLocation() == VariableDeclaration::Location::Storage)
+		for (Type const* subtype: OversizedTypeChecker{*_variable.annotation().type}.oversizedTypes())
+			m_errorReporter.warning(
+				7325_error,
```

Comment:
I reduced this to a single error number and also changed the error message to always talk about "Type" because it would complicate the code a lot otherwise.